File: ios/Modules/Sources/Bootstrap/api/NodePreferences.swift
import Foundation

/// Persistent storage of the user-chosen node URL.
///
/// Narrow on purpose — JWTs and keys live in Keychain (Wave 5). Only
/// what we need to boot: the node base URL.
public protocol NodePreferences: Sendable {
    var baseUrlStream: AsyncStream<String?> { get }
    func currentBaseUrl() -> String?
    func setBaseUrl(_ url: String?)
}

File: android/app/src/main/java/sol/vortexx/android/bootstrap/api/NodePreferences.kt
package sol.vortexx.android.bootstrap.api

import kotlinx.coroutines.flow.Flow

/**
 * Persistent storage of the user-chosen node URL.
 *
 * Narrow on purpose — JWTs / keys live in a separate encrypted store
 * (Wave 5). Only what we need to boot: the node base URL.
 */
interface NodePreferences {
    val baseUrl: Flow<String?>
    suspend fun setBaseUrl(url: String?)
}
